### PR TITLE
Fix for inability to read some parquet files (issue #816)

### PR DESCRIPTION
**Pull request by daw3rd**
## Why are these changes needed?
To allow proper reading and processing of some parquet files that are otherwise unreadable.

## Related issue number (if any).

#816

### CODE REVIEW

File: data-processing-lib/python/src/data_processing/utils/transform_utils.py [hunk 1]
```diff
@@ -144,8 +145,18 @@ def convert_binary_to_arrow(data: bytes, schema: pa.schema = None) -> pa.Table:
             table = pq.read_table(reader, schema=schema)
             return table
         except Exception as e:
-            logger.error(f"Failed to convert byte array to arrow table, exception {e}. Skipping it")
-            return None
+            logger.warning(f"Could not convert bytes to pyarrow: {e}")
+
```
Comment (blublinsky) [inline data-processing-lib/python/src/data_processing/utils/transform_utils.py:149]: Can you, please put a comment here about why polars. Just copy the blur from the where you found this solution

**Review by blublinsky — APPROVED** [review]
Approving, but please, add the comment

File: data-processing-lib/python/src/data_processing/utils/transform_utils.py (hunk 1 above)
Comment (daw3rd) [reply data-processing-lib/python/src/data_processing/utils/transform_utils.py:149]: Done.

File: transforms/universal/filter/python/src/filter_transform.py [hunk 2]
```diff
@@ -67,6 +67,10 @@ def __init__(self, config: dict):
         self.logical_operator = config.get(filter_logical_operator_key, filter_logical_operator_default)
         self.columns_to_drop = config.get(filter_columns_to_drop_key, filter_columns_to_drop_default)
 
+        # Temporarily here to test if this can allow use to process files that are required to be read by polars for mm
```
Comment (touma-I) [inline transforms/universal/filter/python/src/filter_transform.py]: consider removing the 4 commented lines or restore file to its original commit.

File: data-processing-lib/python/requirements.txt [hunk 3]
```diff
@@ -4,3 +4,4 @@
   argparse
   mmh3
   psutil
+  polars
```
Comment (touma-I) [inline data-processing-lib/python/requirements.txt]: If this is the first module to introduce polars, consider adding a >= version constraints

**Review by touma-I — CHANGES_REQUESTED** [review]
Let me know if the two comments can be addressed. Thanks

File: data-processing-lib/python/src/data_processing/utils/transform_utils.py [hunk 4]
```diff
@@ -144,8 +145,21 @@ def convert_binary_to_arrow(data: bytes, schema: pa.schema = None) -> pa.Table:
...
+        # "Nested data conversions not implemented for chunked array outputs"
+        # See issue 816 https://github.com/IBM/data-prep-kit/issues/816.
+        logger.info(f"Attempting read of pyarrow Table using polars")
+        try:
+            import polars
+
+            df = polars.read_parquet(io.BytesIO(data))
+            table = df.to_arrow()
+        except Exception as e:
+            logger.warning(f"Could not convert bytes to pyarrow using polars: {e}. Skipping.")
```
Comment (touma-I) [inline data-processing-lib/python/src/data_processing/utils/transform_utils.py]: should this be logger.error ?

File: data-processing-lib/python/requirements.txt (hunk 3 above)
Comment (daw3rd) [reply data-processing-lib/python/requirements.txt]: set to >=1.16.0

File: data-processing-lib/python/src/data_processing/utils/transform_utils.py (hunk 4 above)
Comment (daw3rd) [reply data-processing-lib/python/src/data_processing/utils/transform_utils.py]: Good catch.  Thanks.

File: transforms/universal/filter/python/src/filter_transform.py (hunk 2 above)
Comment (daw3rd) [reply transforms/universal/filter/python/src/filter_transform.py]: done

**Review by touma-I — APPROVED** [review]
Thanks

**Pull request merged**